### PR TITLE
Improve the Standard Library Full Build Script

### PR DESCRIPTION
**Pull request by ThisaruGuruge**
## Purpose

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/4002

This will improve the Standard Library Build script with the following:

- There was a bug that prevented force-checkout the repositories. It is now fixed.
- When a full build is running, the SNAPSHOT build does not change the version number. It just replace the timestamp version with `SNAPSHOT`. This may cause issues when we build the full standard library using the script.
- Introduce `--from-module` flag: This will help to start the build from a particular module.
- Change the `--module` flag to `--up-to-module` to make the purpose of the flag clear.
- Change the warning color to Yellow from Red.
- Introduce `--continue-on-error` flag and make the default behavior to stop the subsequent builds when a module build is failed. The existing behavior is to continue on error, no matter what.

### CODE REVIEW

File: resources/scripts/build_standard_library.py [hunk 1]
```diff
@@ -208,17 +282,31 @@ def clone_module(module_link):
...
+        for line in properties:
+            if re.match("^version=", line):
+                module_name = module[FIELD_NAME].split("-")[-1]
+                if module_name == "jballerina.java.arrays":
+                    module_name = "JavaArrays"
+                elif module_name == "oauth2":
+                    module_name = "OAuth2"
+                else:
+                    module_name = module_name.capitalize()
+                version_dict[module_name] = line.split("=")[1].strip()
```
Comment (chiranSachintha) [inline resources/scripts/build_standard_library.py:302]: We can use `version_key` to update the dictionary. (The version key of the constant module is  `stdlibConstraintVersion`)
```python
version_dict[module['version_key']] = module['module_version']

```

File: resources/scripts/build_standard_library.py [hunk 2]
```diff
@@ -187,13 +250,24 @@ def replace_lang_version(version):
...
 
-def replace_stdlibs_version():
+def replace_stdlibs_version(module_name, snapshots_build):
+    global version_dict
     with open(TEMP_PROPERTIES, "w+") as temp, open(GRADLE_PROPERTIES) as properties:
         for line in properties:
-            if re.match("^stdlib.*Version=", line):
-                version_string = line.split("=")[0]
-                version_number = line.split("=")[1].split("-")[0]
-                line = version_string + "=" + version_number + "-SNAPSHOT" + "\n"
```
Comment (chiranSachintha) [inline resources/scripts/build_standard_library.py:196]: By keeping it as 'stdlibConstraintVersion', we can directly access the version number from the dictionary using the 'version_string'.
```
line = version_string + "=" + version_dict[version_string]  + "\n"
```

File: resources/scripts/build_standard_library.py [hunk 3]
```diff
@@ -208,17 +282,31 @@ def clone_module(module_link):
...
+                module_name = module[FIELD_NAME].split("-")[-1]
+                if module_name == "jballerina.java.arrays":
+                    module_name = "JavaArrays"
+                elif module_name == "oauth2":
+                    module_name = "OAuth2"
+                else:
+                    module_name = module_name.capitalize()
+                version_dict[module_name] = line.split("=")[1].strip()
+
+
```
Comment (chiranSachintha) [inline resources/scripts/build_standard_library.py:304]: This extra line can be removed

File: resources/scripts/build_standard_library.py [hunk 4]
```diff
@@ -156,26 +216,29 @@ def process_module(module, commands, lang_version, use_snapshots):
...
-        replace_stdlibs_version()
+        replace_stdlibs_version(module[FIELD_NAME], use_snapshots)
 
-    subprocess.run(commands)
+    proc = subprocess.run(commands)
 
     os.chdir("..")
 
+    return proc.returncode
+
```
Comment (chiranSachintha) [inline resources/scripts/build_standard_library.py:241]:
```suggestion
```

File: resources/scripts/build_standard_library.py (hunk 3 above)
Comment (ThisaruGuruge) [reply resources/scripts/build_standard_library.py:296]: These extra lines are added by the Python formatter. (The convention is to use two empty lines between two function definitions).

File: resources/scripts/build_standard_library.py (hunk 1 above)
Comment (ThisaruGuruge) [reply resources/scripts/build_standard_library.py]: Fixed with 33d788f226ccf744b0fe702ece2da53c1259f4f7

File: resources/scripts/build_standard_library.py (hunk 2 above)
Comment (ThisaruGuruge) [reply resources/scripts/build_standard_library.py]: Fixed with 33d788f226ccf744b0fe702ece2da53c1259f4f7

**Review by niveathika — APPROVED** [review]

**Pull request merged**